android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/translate/GeckoTranslationUtils.kt

```diff
@@ -0,0 +1,61 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package mozilla.components.browser.engine.gecko.translate
+
+import mozilla.components.concept.engine.translate.TranslationError
+import org.mozilla.geckoview.TranslationsController.TranslationsException
+
+/**
+ * Utility file for translations functions related to the Gecko implementation.
+ */
+object GeckoTranslationUtils {
+    /**
+     * Convenience method for mapping a [TranslationsException] to the Android Components defined
+     * error type of [TranslationError].
+     *
+     * @param throwable The engine throwable that occurred during translating. Ordinarily should be
+     * a [TranslationsException].
+     */
+    fun fromGeckoErrorToTranslationError(throwable: Throwable): TranslationError =
+        if (throwable is TranslationsException) {
```

Comment:
@ohall-m this is styling feedback that I only thought off a bit too late, so feel free to consider it as a future change or ignore!

In Kotlin, you can add type-safe utility methods with extension functions that are identical to this as static java methods, but might be easier to use:

```kotlin
fun TranslationsException.intoTranslationError(): TranslationError {
  return when (code) {
    // ...
  }
}
```

This could mean one less type-check if you add it straight at the caller.

Reply:
Ohh, thank you, that's neat! I miss out on a lot of cool Kotlin features. 

I'll be revisiting in [bug 1872703](https://bugzilla.mozilla.org/show_bug.cgi?id=1872703) to connect this to all the translations calls. 

I'll adjust it over there very shortly since this bug just landed. I'll leave a note on the bug!